xonsh/parser.py

```diff
@@ -333,15 +331,18 @@ def expr(self, p):
 
     def token_col(self, t):
         """Gets ths token column"""
-        return self.lexer.token_col(t)
+        return t.lexpos
 
     @property
     def lineno(self):
-        return self.lexer.lineno
+        try:
+            return self.lexer.last.lineno
+        except:
+            return 0
```

Comment:
I don't know what the exception is here, but it is probably best to use `if hasattr(last, ..)` or `if last is None:` or similar.